speech/src/main/java/org/openbase/bco/psc/speech/SpeechController.java

```diff
@@ -0,0 +1,290 @@
+        } else if (event.getData() instanceof SpeechHypotheses) {
+            SpeechHypotheses speechHypotheses = (SpeechHypotheses) event.getData();
+            speechHypothesis = speechHypotheses.getBestResult();
+        }
+
+        LOGGER.info("SpeechHypothesis detected: " + speechHypothesis);
+        if (speechHypothesis == null) return;
+
+        try {
+            // try mapping speech hypothesis to action parameter
+            ActionParameter actionParameter = keywordConverter.getAction(speechHypothesis);
+            if (actionParameter == null) {
+                LOGGER.warn("No matching action found.");
+            }
+
+            // send action parameter to Control component
+            rsbConnection.publishData(actionParameter);
+            LOGGER.info("PUBLISHED action: " + actionParameter);
+
+        } catch (CouldNotPerformException ex) {
+            ExceptionPrinter.printHistory(ex, LOGGER, LogLevel.ERROR);
+        } catch (InterruptedException ex) {
+            Thread.currentThread().interrupt();
+            ExceptionPrinter.printHistory(ex, LOGGER, LogLevel.ERROR);
+        }
+
+    }
+
+    private void initializeRegistryConnection() throws InterruptedException, CouldNotPerformException {
+        try {
+            LOGGER.info("Waiting for bco registry synchronization...");
+            Registries.getUnitRegistry().waitForData();
+
+        } catch (NotAvailableException ex) {
+            throw new CouldNotPerformException("Could not connect to the registry.", ex);
+        } catch (CouldNotPerformException ex) {
+            throw new CouldNotPerformException("The RegistrySynchronization could not be activated although connection to the registry is possible.", ex);
+        }
+    }
+
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void init() throws InitializationException, InterruptedException {
+        LOGGER.info("Initializing speech controller...");
+        if (!initialized) {
+            try {
+                initKeywordConverter();
+
+                initializeRegistryConnection();
+                rsbConnection = new RSBConnection(this);
+                rsbConnection.init();
+                initialized = true;
+            } catch (CouldNotPerformException ex) {
+                throw new InitializationException(SpeechController.class, ex);
+            } catch (IOException ex) {
+                ExceptionPrinter.printHistory(ex, LOGGER, LogLevel.ERROR);
+            }
+        }
+    }
+
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void activate() throws CouldNotPerformException, InterruptedException {
+        LOGGER.debug("Activating " + getClass().getName() + ".");
+        if (!initialized) {
+            throw new CouldNotPerformException("Activate can only be called after init.");
+        }
+        if (!active) {
+            active = true;
+            Registries.waitForData();
```

Comment:
Since this can take a while, it would be useful to previously print an INFO like "waiting for bco registry..."